templates/queries.yaml.j2

```
@@ -214,3 +214,195 @@ pg_statio_all_tables:
+      datname,
+      query,
+      queryid,
+      calls,
+      total_time,
+      rows,
+      shared_blks_hit, shared_blks_read, shared_blks_dirtied, shared_blks_written,
+      local_blks_hit, local_blks_read, local_blks_dirtied, local_blks_written,
+      temp_blks_read, temp_blks_written,
+      blk_read_time, blk_write_time
+    FROM pg_stat_statements
+    INNER JOIN pg_database ON (dbid = pg_database.oid)
+    INNER JOIN pg_user ON (userid = pg_user.usesysid);
+  metrics:
+    - usename:
+        usage: "LABEL"
+        description: "User who executed the statement"
+    - datname:
+        usage: "LABEL"
+        description: "Database in which the statement was executed"
+    - query:
+        usage: "LABEL"
+        description: "Text of a representative statement"
+    - queryid:
+        usage: "LABEL"
+        description: "Query ID"
+    - calls:
+        usage: "COUNTER"
+        description: "Number of times executed"
+    - total_time:
+        usage: "COUNTER"
+        description: "Total time spent in the statement, in milliseconds"
+    - rows:
+        usage: "COUNTER"
+        description: "Total number of rows retrieved or affected by the statement"
+    - shared_blks_hit:
+        usage: "COUNTER"
+        description: "Total number of shared block cache hits by the statement"
+    - shared_blks_read:
+        usage: "COUNTER"
+        description: "Total number of shared blocks read by the statement"
+    - shared_blks_dirtied:
+        usage: "COUNTER"
+        description: "Total number of shared blocks dirtied by the statement"
+    - shared_blks_written:
+        usage: "COUNTER"
+        description: "Total number of shared blocks written by the statement"
+    - local_blks_hit:
+        usage: "COUNTER"
+        description: "Total number of local block cache hits by the statement"
+    - local_blks_read:
+        usage: "COUNTER"
+        description: "Total number of local blocks read by the statement"
+    - local_blks_dirtied:
+        usage: "COUNTER"
+        description: "Total number of local blocks dirtied by the statement"
+    - local_blks_written:
+        usage: "COUNTER"
+        description: "Total number of local blocks written by the statement"
+    - temp_blks_read:
+        usage: "COUNTER"
+        description: "Total number of temp blocks read by the statement"
+    - temp_blks_written:
+        usage: "COUNTER"
+        description: "Total number of temp blocks written by the statement"
+    - blk_read_time:
+        usage: "COUNTER"
+        description: "Total time the statement spent reading blocks, in milliseconds (if track_io_timing is enabled, otherwise zero)"
+    - blk_write_time:
+        usage: "COUNTER"
+        description: "Total time the statement spent writing blocks, in milliseconds (if track_io_timing is enabled, otherwise zero)"
+
+
+pg_stat_activity_state:
+  query: |
+    SELECT
+      usename,
+      datname,
+      application_name,
+      state,
+      COUNT(*)
```

```suggestion
      COUNT(*) as total
```

Do you need something like this in order to reference 'total' below?

---

I don't think so, as you give it a list, but may as well anyway